32blit-pico/led.cpp

```diff
@@ -13,6 +13,24 @@
 #if defined(LED_R_PIN) && defined(LED_G_PIN) && defined(LED_B_PIN)
 static const int led_pins[]{LED_R_PIN, LED_G_PIN, LED_B_PIN};
 #define HAVE_LED
+#elif defined(LED_WS2812_PIN)
+#include "hardware/pio.h"
+#include "ws2812.pio.h"
+
+static uint32_t last_color = 0;
+static int pio_sm = -1;
+
+static void put_pixel(uint8_t r, uint8_t g, uint8_t b, uint8_t a) {
+  if(pio_sm < 0) return;
+  uint32_t color =
+          (((uint32_t)((r * a) >> 8) << 8)
+          | ((uint32_t)((g * a) >> 8)  << 16)
+          | (uint32_t)((b * a) >> 8)) << 8u;
```

Comment:
Not sure about having the alpha channel do something here when it doesn't for "regular" RGB LEDs...

Reply:
This is really working (tested), colors are "dimmed" i would say, sending 0 as alpha will completely turn off the led.

Reply:
My point was more that it's inconsistent to do that for one type of LED but not the other (and also the other port that has an LED).

But doing it everywhere would be a bigger change... and entirely independent from adding a new driver...